### ISSUE TITLE
[RLlib] Can't restore checkpoint that saved from another computer

### ISSUE THREAD

**Issue by palkan21** [ray-project/ray#19804]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### What happened + What you expected to happen

I want to save checkpoint at first computer, and restore checkpoint from second computer.
Save checkpoint and load it directly on same computer is walking, 
but I can't do the same job when using two computers.
After save checkpoint, I transfer all the file created from first computer to second computer.

Traceback (most recent call last):
  File "muscle_network_train.py", line 2177, in <module>
    main()
  File "muscle_network_train.py", line 2001, in main
    agent.restore('output_rllib/checkpoint_000150/checkpoint-150') #pickle m
  File "/home/mk/.local/lib/python3.8/site-packages/ray/tune/trainable.py", line 377, in restore
    self.load_checkpoint(checkpoint_path)
  File "/home/mk/.local/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 785, in load_checkpoint
    self.__setstate__(extra_data)
  File "/home/mk/.local/lib/python3.8/site-packages/ray/rllib/agents/trainer_template.py", line 235, in __setstate__
    Trainer.__setstate__(self, state)
  File "/home/mk/.local/lib/python3.8/site-packages/ray/rllib/agents/trainer.py", line 1545, in __setstate__
    self.workers.local_worker().restore(state["worker"])
  File "/home/mk/.local/lib/python3.8/site-packages/ray/rllib/evaluation/rollout_worker.py", line 1233, in restore
    objs = pickle.loads(objs)
AttributeError: Can't get attribute 'PolicySpec' on <module 'ray.rllib.policy.policy' from '/home/mk/.local/lib/python3.8/site-packages/ray/rllib/policy/policy.py'>

### Versions / Dependencies

Ray 1.5.1
Python 3.8.10
Ubuntu 20.04

### Reproduction script

temp_dir = os.path.abspath(os.getcwd())
ray.init(_temp_dir=temp_dir, num_cpus=0, num_gpus=0, ignore_reinit_error=True, log_to_driver=False)
#save model
agent = PPOTrainer(config, TestEnvMuscle)  #TestEnvMuscle is my own class
result = agent.train()
checkpoint_path = agent.save('output_rllib/')

#restore model
agent = PPOTrainer(config, TestEnvMuscle)
agent.restore('output_rllib/checkpoint_000150/checkpoint-150')

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by Dylan-Kerler** [comment]
@palkan21 Did you fix this issue? I'm getting a very similar error

**Comment by Dylan-Kerler** [comment]
I thought it might help if I downgraded from python3.8 to 3.7 but then I get this error instead:

```
line 1096, in restore
    objs = pickle.loads(objs)
ValueError: unsupported pickle protocol: 5
```

**Comment by avnishn** [comment]
can you both provide me with scripts for reproducing your issues. If its possible to post them in google colab links that would also be great.

**Comment by Sridhar701Pitt** [comment]
[checkpoint_010000.zip](https://github.com/ray-project/ray/files/8190289/checkpoint_010000.zip)
I have the same issue. Attached is my checkpoint file. Restoring this give me the following error.
```
Traceback (most recent call last):
  File "examples/hover_learn_gcp_sac.py", line 62, in <module>
    agent.restore("/root/Python_ws/results/no_fail_sac_hover_1/SACTrainer_hover-aviary-v0_2022-03-04_08-11-26re34t9x6/checkpoint_010000/checkpoint-10000")
  File "/usr/local/lib/python3.8/dist-packages/ray/tune/trainable.py", line 467, in restore
    self.load_checkpoint(checkpoint_path)
  File "/usr/local/lib/python3.8/dist-packages/ray/rllib/agents/trainer.py", line 1823, in load_checkpoint
    self.__setstate__(extra_data)
  File "/usr/local/lib/python3.8/dist-packages/ray/rllib/agents/trainer.py", line 2443, in __setstate__
    self.workers.local_worker().restore(state["worker"])
  File "/usr/local/lib/python3.8/dist-packages/ray/rllib/evaluation/rollout_worker.py", line 1326, in restore
    objs = pickle.loads(objs)
TypeError: an integer is required (got type bytes)
```

 I trained my program using SACTrainer in google cloud (GCP) inside a custom docker container. It might be that my current config for the trainer is different from the config when I ran this program (whose values i don't remember/have) but I am not sure.

Any help would be greatly appreciated (even just knowing the number and type of the observation and action spaces might work).

**Comment by avnishn** [comment]
Is your ray version the same on your pc as it is from your gcp experiment? They'll need to be the same version.

**Comment by Sridhar701Pitt** [comment]
I checked and both run ray v1.10.0

**Comment by Sridhar701Pitt** [comment]
Not sure what the issue is but I had to use slightly different containers for running on the GCP and on the local machine. Both containers have the same Ray version (1.10.0). I was able to restore the agents only in the containers the checkpoint file was generated from. The main application script run with these containers is exactly the same. 

I suspect it might be another dependency (other than Ray) that might be of issue but not sure. Attached is a comparison(diff) screenshot of $pip list from both containers. Tool: text-compare.com

![Screenshot 2022-03-05 at 21-45-18 Gmail - Your comparison from Text Compare ](https://user-images.githubusercontent.com/56744642/156907134-6ac894c4-e8f1-4109-a58e-a6233fee6b7f.png)
